From 888db0f8dea684fddafdfdfdab51cdfa2d933a37 Mon Sep 17 00:00:00 2001
From: Antoine Tenart <antoine.tenart@bootlin.com>
Date: Wed, 13 Feb 2019 13:52:15 +0100
Subject: [PATCH 115/386] net: mvpp2: make phylink validate GMII aware

This patch adds (back) support for GMII in the phylink validate helper
in the Marvell PPv2 driver.

Change-Id: I4845bfa02a7d04c743a83dc430fbeefece21d7b0
Signed-off-by: Antoine Tenart <antoine.tenart@bootlin.com>
Reviewed-on: https://sj1git1.cavium.com/6483
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
index aa6632ab160a..383c4deb7ef1 100644
--- a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
@@ -5826,6 +5826,7 @@ static void mvpp2_phylink_validate(struct net_device *dev,
 		if (port->gop_id != 0)
 			goto empty_set;
 		break;
+	case PHY_INTERFACE_MODE_GMII:
 	case PHY_INTERFACE_MODE_RGMII:
 	case PHY_INTERFACE_MODE_RGMII_ID:
 	case PHY_INTERFACE_MODE_RGMII_RXID:
@@ -5860,6 +5861,7 @@ static void mvpp2_phylink_validate(struct net_device *dev,
 		if (port->has_xlg_mac)
 			phylink_set(mask, 5000baseT_Full);
 		/* Fall-through */
+	case PHY_INTERFACE_MODE_GMII:
 	case PHY_INTERFACE_MODE_RGMII:
 	case PHY_INTERFACE_MODE_RGMII_ID:
 	case PHY_INTERFACE_MODE_RGMII_RXID:
-- 
2.17.1

